### PR TITLE
Google Login

### LINKED ISSUE #861: Google log in
Allow log in with Google and show 'Log in with Google' at the bottom of the log in screen. Show this both in email and password state.

Zeplin: https://zpl.io/b6JBDlj

toggl - ios - log in step1 1x

### PR DESCRIPTION
**Pull request by willsb**
Closes #861
Ref #862 

To do a functional test you need to:

- Replace the variables locally (done easily by running the cake script) 
- Add the `USE_ANALYTICS` symbol to debug builds

~Until @amulware updates the provisioning profiles with the appropriate entitlements, this can only be tested on simulators.~

~Mergin this without the proper provisioning profiles **will break building for devices**, so please don’t squash until this is solved.~

Turns out the above was not needed 😄

### CODE REVIEW

**Comment by amulware** [comment]
I will take care of the provisioning tomorrow.

File: Toggl.Daneel/Services/GoogleService.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+        [Export("signIn:dismissViewController:")]
+        public void DismissViewController(SignIn signIn, UIViewController viewController)
+        {
+            var presenter = Mvx.Resolve<IMvxIosViewPresenter>() as MvxIosViewPresenter;
+            presenter.MasterNavigationController.DismissViewController(true, null);
+        }
+
+        [Export("signInWillDispatch:error:")]
+        public void WillDispatch(SignIn signIn, NSError error)
```
Comment (arturs105) [inline Toggl.Daneel/Services/GoogleService.cs]: Is this necessary?

File: Toggl.Foundation.Tests/Login/LoginManagerTests.cs [hunk 2]
```diff
@@ -294,40 +296,100 @@ public void ThrowsIfYouPassInvalidParameters(string password)
...
             [Fact, LogIfTooSlow]
             public async Task CallsTheGetMethodOfTheUserApi()
             {
                 await LoginManager.RefreshToken(Password);
+ 
+                 await Api.User.Received().Get();
+            }
+ 
+            [Fact, LogIfTooSlow]
+            public async Task ShouldPersistTheUserToTheDatabase()
```
Comment (arturs105) [inline Toggl.Foundation.Tests/Login/LoginManagerTests.cs]: The naming for this (and the next 2 tests) is inconsistent. Usually the names are:
```
class TheDoSomenthingMethod
{
    public void DoesStuff() {} //not ShouldDoStuff
}
```

File: Toggl.Foundation/Exceptions/GoogleLoginException.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+namespace Toggl.Foundation.Exceptions
+{
+    public class GoogleLoginException : Exception
```
Comment (arturs105) [inline Toggl.Foundation/Exceptions/GoogleLoginException.cs]: Class not sealed

File: Toggl.Ultrawave/Network/Credentials.cs [hunk 4]
```diff
@@ -1,48 +1,57 @@
...
-            Ensure.Argument.IsNotNull(authString, nameof(authString));
-
-            var authStringBytes = Encoding.UTF8.GetBytes(authString);
-            var authHeader = Convert.ToBase64String(authStringBytes);
-
-            return new HttpHeader("Authorization", authHeader, HttpHeader.HeaderType.Auth);
-        }
-    }
-}
+﻿using System;
```
Comment (arturs105) [inline Toggl.Ultrawave/Network/Credentials.cs]: What the hell has happened in this file?

Comment (willsb) [reply Toggl.Ultrawave/Network/Credentials.cs]: crlf

**Comment by amulware** [comment]
Please update the description above or leave a comment with new findings before closing this for future prosperity.

**Pull request merged**